.github/workflows/release.yml

```
@@ -146,45 +165,81 @@ jobs:
 
       - name: Install
         run: |
-          bin=$(cabal -v0 --project-file=cabal.static.project list-bin print-api)
           mkdir distribution
-          cp ${bin} distribution/print-api
+          for executable in 'print-api' 'print-api-${{ matrix.ghc }}'; do
+            path=$(cabal -v0 --project-file=cabal.static.project list-bin "print-api:exe:${executable}")
+            cp "${path}" distribution
+          done
 
-      - name: File type
-        run: file distribution/print-api
+      - name: File type of executables
+        run: file distribution/*
+
+      - name: Post-process executables
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        run: |
+          for executable in distribution/*; do
+            strip "${executable}"
+            upx -9 "${executable}"
+          done
 
-      - name: Package the print-api executable
+      - name: Package the executables
+        shell: bash
         run: |
-          PRINTAPI_EXEC=distribution/print-api
-          .github/workflows/process-binaries.sh
-          DIR=$(dirname $PRINTAPI_EXEC)
-          FILE=$(basename $PRINTAPI_EXEC)
+          executables=( $(ls distribution) )
           version=$(./distribution/print-api --version)
-          PRINTAPI_EXEC_TAR=print-api-${version}-${{ runner.os }}-static-${{ matrix.ghc }}-${{ env.ARCH }}.tar.gz
-          tar -czvf $PRINTAPI_EXEC_TAR -C $DIR $FILE
-          echo PRINTAPI_EXEC_TAR=$PRINTAPI_EXEC_TAR >> $GITHUB_ENV
+          archive="print-api-${version}-${{ runner.os }}-${{ env.ARCH }}-static-ghc-${{ matrix.ghc }}.tar.gz"
+          tar -c -z -v -f "${archive}" -C distribution "${executables[@]}"
+          echo "PRINTAPI_EXEC_TAR=${archive}" >> $GITHUB_ENV
 
-      - name: Upload print-api executable to workflow artifacts
+      - name: Upload the executables archive
         uses: actions/upload-artifact@v3
         with:
           name: artifact
           path: ${{ env.PRINTAPI_EXEC_TAR }}
 
+  # NOTE: We run upx on the executables in the tests/build-alpine job.
+  # That means, the assets that are linked dynamically _appear_ to be statically linked after that.
```

That's not quite the correct phrasing, I would rather say:
```suggestion
  # NOTE: We run upx on the executables in every linux job to compress them.
  # That means, the assets that are linked dynamically _appear_ to be statically linked after that, despite not being compiled statically in Alpine with musl. They are still dynamically-linked once they are decompresed.
```